controllers/nginx/pkg/config/config.go

```diff
@@ -307,20 +301,14 @@ func NewDefault() Configuration {
 	return cfg
 }
 
-// BuildLogFormatUpstream format the log_format upstream based on proxy_protocol
-func BuildLogFormatUpstream(useProxyProtocol bool, curLogFormatUpstream string) string {
-
-	// test if log_format comes from configmap
-	if curLogFormatUpstream != "" &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr") &&
-		curLogFormatUpstream != fmt.Sprintf(logFormatUpstream, "$remote_addr") {
-		return curLogFormatUpstream
-	}
-
-	if useProxyProtocol {
-		return fmt.Sprintf(logFormatUpstream, "$proxy_protocol_addr")
+// BuildLogFormatUpstream format the log_format upstream using
+// proxy_protocol_addr as remote client address if UseProxyProtocol
+// is enabled.
+func (cfg Configuration) BuildLogFormatUpstream() string {
+	if cfg.UseProxyProtocol {
+		return fmt.Sprintf("$proxy_protocol_addr - %s", cfg.LogFormatUpstream)
```

Comment:
This is breaking overrides of the log format and just seems extremely sloppy as users are now stuck with a hardcoded string format (`$ip - $customlogformat`) regardless of custom logging formats.

Please revert or fix this as i (and i figure lots of people) want to define my own *structured* logging format and this IP prefix is preventing this.